### ISSUE TITLE
PyPy 2.5.1 installation with python-build fails on Trusty

### ISSUE THREAD

**Issue by BanzaiMan** [yyuu/pyenv#366]
This works on Precise, but fails on Trusty.

```sh-session
travis@trusty64:~$ python-build pypy-2.5.1 /opt/python/pypy-2.5.1

ERROR: The binary distribution of PyPy is not available for linux64.
try 'pypy-2.5.1-src' to build from soruce.
```

Note that this failure does not stop Chef, since:

```sh-session
travis@trusty64:~$ echo $?
0
```

And building from source also fails:

```
travis@trusty64:~$ python-build pypy-2.5.1-src /opt/python/pypy-2.5.1
Downloading release-2.5.1.tar.bz2...
-> https://bitbucket.org/pypy/pypy/get/release-2.5.1.tar.bz2

BUILD FAILED (Ubuntu 14.04 using python-build 20141028)

Inspect or clean up the working tree at /tmp/python-build.20150425120047.17681
Results logged to /tmp/python-build.20150425120047.17681.log

Last 10 log lines:
/tmp/python-build.20150425120047.17681 ~

checksum mismatch: pypy-pypy-9c4588d731b7.tar.bz2 (file is corrupt)
expected 58cefb30a645f7b535e2b63a1d7c21bf12615ad95324bc6c919df90f7b2f37ef, got 32b33d407797f78d9453c81f2066c981501beddc17eaf8f7bcdc1798017bf03f
```

On the other hand, pypy-2.5.0 can be installed successfully on Trusty.

**Comment by yyuu** [comment]
Looks weird. The definition of `pypy-2.5.1` seems to have case clause for `linux64` architecture. It should not go to _else_ clause.

https://github.com/yyuu/pyenv/blob/50d5de473168ba159b9035ee592a937c68461bf3/plugins/python-build/share/python-build/pypy-2.5.1#L21

And, even if it goes to the _else_ clause, it should exit with 1, not 0.

https://github.com/yyuu/pyenv/blob/50d5de473168ba159b9035ee592a937c68461bf3/plugins/python-build/share/python-build/pypy-2.5.1#L42

**Comment by yyuu** [comment]
I couldn't reproduce the behaviour on a Docker container of Ubuntu trusty.

```
root@1d149f807681:~# pyenv --version
pyenv 20150326-485-g50d5de4
root@1d149f807681:~# uname -a
Linux 1d149f807681 3.16.0-4-amd64 #1 SMP Debian 3.16.7-ckt9-2 (2015-04-13) x86_64 x86_64 x86_64 GNU/Linux
root@1d149f807681:~# lsb_release -a 
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 14.04.2 LTS
Release:        14.04
Codename:       trusty
root@1d149f807681:~# pyenv --version
pyenv 20150326-485-g50d5de4
root@1d149f807681:~# pyenv install pypy-2.5.1
Downloading pypy-2.5.1-linux64.tar.bz2...
-> https://bitbucket.org/pypy/pypy/downloads/pypy-2.5.1-linux64.tar.bz2
Installing pypy-2.5.1-linux64...
Installing setuptools from https://bootstrap.pypa.io/ez_setup.py...
Installing pip from https://bootstrap.pypa.io/get-pip.py...
Installed pypy-2.5.1-linux64 to /root/.pyenv/versions/pypy-2.5.1

root@1d149f807681:~# pyenv versions
* system (set by /root/.pyenv/version)
  pypy-2.5.1
root@1d149f807681:~# pyenv shell pypy-2.5.1
root@1d149f807681:~# python --version
Python 2.7.9 (9c4588d731b7, Mar 23 2015, 16:30:30)
[PyPy 2.5.1 with GCC 4.6.3]
```

**Comment by BanzaiMan** [comment]
Thanks for looking into it. I'll double check here and report back.

**Comment by joshfriend** [comment]
@BanzaiMan I believe I fixed this issue in #347. The URL your version of pyenv is downloading matches the old version of what was changed in the PR.
